tests/image-test.js

```diff
@@ -0,0 +1,104 @@
+// var resemble = require('node-resemble');
+var $ = require('jquery');
+
+module.exports = {};
+
+/* Compare an image from a canvas object by sending it to our express server
+ * letting it do the work.
+ *
+ * @param {string} name: name of the base image.  This is probably the name of
+ *    the test.
+ * @param {HTMLCanvasObject} canvas: the image is obtained via
+ *    canvas.toDataURL().
+ * @param {number} threshold: allowed difference between this image and the
+ *    base image.
+ * @param {function} callback: a function to call when complete.
+ */
+function compareImage(name, canvas, threshold, callback) {
+  if (threshold === undefined) {
+    threshold = 0.001;
+  }
+  $.ajax({
+    url: '/testImage?compare=true&threshold=' + encodeURIComponent(threshold) + '&name=' + encodeURIComponent(name),
+    data: '' + canvas.toDataURL(),
+    method: 'PUT',
+    contentType: 'image/png',
+    dataType: 'json'
+  }).success(function (data) {
+    expect(Number(data.misMatchPercentage) * 0.01).not.toBeGreaterThan(threshold);
+    if (callback) {
+      callback();
+    }
+  });
+}
+
+/**
+ * Call this before creating any geojs canvas instances to ensure that the
+ * image test can collect data.
+ */
+module.exports.prepareImageTest = function () {
+  window.contextPreserveDrawingBuffer = true;
+  $('#map').remove();
+  var map = $('<div id="map"/>').css({width: '800px', height: '600px'});
+  $('body').css({overflow: 'hidden'}).append(map);
+};
+
+/**
+ * Compare a composite of all canvas elements with a base image.
+ *
+ * @param {string} name: name of the base image.  This is probably the name of
+ *    the test.
+ * @param {number} threshold: allowed difference between this image and the
+ *    base image.
+ * @param {function} doneFunc: a function to call when complete.  Optional.
+ * @param {function} idleFunc: a function to call to ensure that the map is
+ *    idle.  No other tests or delays are supplied until this is called.  It
+ *    is expected to take a callback function itself.
+ * @param {number} delay: additional delay in milliseconds to wait after idle.
+ * @param {integer} rafCount: additional number of renderAnimationFrames to
+ *    wait after the delay.
+ */
+module.exports.imageTest = function (name, threshold, doneFunc, idleFunc, delay, rafCount) {
```

Comment:
I think this would be much simpler expressed as promises.  For the idleFunc, couldn't we just have the person running the test call when it is ready.

```javascript
setupTestCase()
  .then(imageTest)
  .then(doSomethingAfter)
  .catch(handleErrors);
```